From bc46b24f2205b107cb0ff134bcd0a0f6c593f29e Mon Sep 17 00:00:00 2001
From: CanYang He <hcy@rock-chips.com>
Date: Wed, 24 Apr 2019 21:43:49 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3399: lp4 frequency change to 416MHz
 856MHz

1. use 856MHz in order to enable write dq odt, because Samsung lp4 write
dq odt can not enable when frequency under 800MHz
2. 416MHz and 856MHz is the frequency avoid disturb wifi 2.4GHz

Change-Id: Icbcd2a78dbcbfe33bc3e5b0e296913dea4b28480
Signed-off-by: CanYang He <hcy@rock-chips.com>
---
 .../rockchip/rk3399-evb-rev3-android-lp4.dts  | 36 ++++++++++++-------
 1 file changed, 23 insertions(+), 13 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android-lp4.dts b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android-lp4.dts
index 581e80a970fd..bcb5a8daf25c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android-lp4.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev3-android-lp4.dts
@@ -114,23 +114,23 @@
 	center-supply = <&vdd_center>;
 	system-status-freq = <
 		/*system status         freq(KHz)*/
-		SYS_STATUS_NORMAL       800000
-		SYS_STATUS_REBOOT       400000
-		SYS_STATUS_SUSPEND      400000
-		SYS_STATUS_VIDEO_1080P  400000
-		SYS_STATUS_VIDEO_4K     800000
-		SYS_STATUS_VIDEO_4K_10B 800000
-		SYS_STATUS_PERFORMANCE  800000
-		SYS_STATUS_BOOST        800000
-		SYS_STATUS_DUALVIEW     800000
-		SYS_STATUS_ISP          800000
+		SYS_STATUS_NORMAL       856000
+		SYS_STATUS_REBOOT       416000
+		SYS_STATUS_SUSPEND      416000
+		SYS_STATUS_VIDEO_1080P  416000
+		SYS_STATUS_VIDEO_4K     856000
+		SYS_STATUS_VIDEO_4K_10B 856000
+		SYS_STATUS_PERFORMANCE  856000
+		SYS_STATUS_BOOST        856000
+		SYS_STATUS_DUALVIEW     856000
+		SYS_STATUS_ISP          856000
 	>;
 	vop-bw-dmc-freq = <
 	/* min_bw(MB/s) max_bw(MB/s) freq(KHz) */
-		0       577      400000
-		578     99999    800000
+		0       577      416000
+		578     99999    856000
 	>;
-	auto-min-freq = <400000>;
+	auto-min-freq = <416000>;
 	auto-freq-en = <1>;
 };
 
@@ -150,6 +150,11 @@
 	opp-400000000 {
 		opp-hz = /bits/ 64 <400000000>;
 		opp-microvolt = <900000>;
+		status = "disabled";
+	};
+	opp-416000000 {
+		opp-hz = /bits/ 64 <416000000>;
+		opp-microvolt = <900000>;
 	};
 	opp-528000000 {
 		opp-hz = /bits/ 64 <528000000>;
@@ -164,6 +169,11 @@
 	opp-800000000 {
 		opp-hz = /bits/ 64 <800000000>;
 		opp-microvolt = <900000>;
+		status = "disabled";
+	};
+	opp-856000000 {
+		opp-hz = /bits/ 64 <856000000>;
+		opp-microvolt = <900000>;
 	};
 	opp-928000000 {
 		opp-hz = /bits/ 64 <928000000>;
-- 
2.35.3

